src/AzureExtension/Widgets/AzurePullRequestsDeveloperWidget.cs

```diff
@@ -0,0 +1,193 @@
+        // data was unavailable for whatever reason.
+        LoadContentData();
+    }
+
+    protected override void ResetDataFromState(string data)
+    {
+        // This widget has no state.
+        return;
+    }
+
+    public override string GetConfiguration(string data)
+    {
+        // This widget has no configuration.
+        return string.Empty;
+    }
+
+    public override void LoadContentData()
+    {
+        if (_cacheManager.NeverUpdated)
+        {
+            SetLoading();
+            return;
+        }
+
+        try
+        {
+            // This can throw if DataStore is not connected.
+            var pullRequestsList = DataManager!.GetPullRequestsForLoggedInDeveloperIds();
+            var itemsData = new JsonObject();
+            var itemsArray = new JsonArray();
+
+            foreach (var pullRequests in pullRequestsList)
+            {
+                var pullRequestsResults = pullRequests is null ? [] : JsonConvert.DeserializeObject<Dictionary<string, object>>(pullRequests.Results);
+                foreach (var element in pullRequestsResults!)
+                {
+                    var workItem = JsonObject.Parse(element.Value.ToStringInvariant());
+
+                    if (workItem != null)
+                    {
+                        // If we can't get the real date, it is better better to show a recent
+                        // closer-to-correct time than the zero value decades ago, so use DateTime.UtcNow.
+                        var dateTicks = workItem["CreationDate"]?.GetValue<long>() ?? DateTime.UtcNow.Ticks;
+                        var dateTime = dateTicks.ToDateTime();
+                        var creator = DataManager.GetIdentity(workItem["CreatedBy"]?.GetValue<long>() ?? 0L);
+                        var item = new JsonObject
+                        {
+                            { "title", workItem["Title"]?.GetValue<string>() ?? string.Empty },
+                            { "url", workItem["HtmlUrl"]?.GetValue<string>() ?? string.Empty },
+                            { "status_icon", GetIconForPullRequestStatus(workItem["PolicyStatus"]?.GetValue<string>()) },
+                            { "number", element.Key },
+                            { "date", TimeSpanHelper.DateTimeOffsetToDisplayString(dateTime, Log) },
+                            { "dateTicks", dateTicks },
+                            { "user", creator.Name },
+                            { "branch", workItem["TargetBranch"]?.GetValue<string>().Replace("refs/heads/", string.Empty) },
+                            { "avatar", creator.Avatar },
+                        };
+
+                        itemsArray.Add(item);
+                    }
+                }
+            }
+
+            // Sort all pull requests by creation date, descending so newer are at the top of the list.
+            var sortedItems = itemsArray.OrderByDescending(x => x?["dateTicks"]?.GetValue<long>());
```

Comment:
NIT:
itemsArray.Where(x => x is not null).OrderByDescending(x => blah).Select(x => x.DeepClone);

or itemsArray.Where(x => x is not null).Select(x => x.DeepClone).OrderByDescending(x => blah)

Reply:
Used the null filter, but kept the list population as-is. I generally don't like using complex expressions to keep the code maintainable and easier to debug.